### ISSUE TITLE
binaries created in releases are not named correctly

### ISSUE THREAD

**Issue by rberlind** [Mastercard/terraform-provider-restapi#2]
I believe that the provider binaries that have been generated under releases are not named correctly. In particular, when I added the v0.2.0 binary, terraform-provider-restapi-darwin-amd64, to my repository under terraform.d/plugins/darwin_amd64, I got errors when I ran `terraform init`.

I was able to successfully initiailize my Terraform code after changing the name of the binary in that directory to "terraform-provider-restapi_v0.2.0-darwin-amd64" in which I have added the version between the name of the provider and the platform.

**Comment by DRuggeri** [comment]
Thanks for the report. Can you elaborate a bit on the workflow that started the error?

Does it sound a bit like this?

1. Install binary in usual plugin location
2. Perform `terraform init` in working location
3. Install new binary (overwriting the old one) in usual plugin location
4. Perform another `terraform init`

Renaming the binary is trivial and something easy to do - I'd just like to understand how the problem cropped up.

**Comment by rberlind** [comment]
I could not get terraform init to run with the binaries named the way they are, at least not on Mac OS X.
I first tried "terraform init" and saw:

Provider "restapi" not available for installation.
A provider named "restapi" could not be found in the official repository.

I then tried "terraform init -plugin-dir=terraform.d" and got:

Missing required providers.

The following provider constraints are not met by the currently-installed
provider plugins:
* restapi (any version)

This was with this code:

provider "restapi" {
  uri = "https://myapi.env.local/api/v1"
}

Of course, I know that is just a placeholder URI, but after changing the binary name, I was able to run "terraform init" with that.

**Comment by DRuggeri** [comment]
Thanks, that helps. I've updated the [README](https://github.com/Mastercard/terraform-provider-restapi/blob/master/README.md) with some installation instructions. I acknowledge the files as they are named cannot be installed and used because of OS and ARCH in the filename. I would rather have subdirectories with OS and ARCH, but github doesn't seem to support that on the release object.

The only option I'm aware of aside from renaming the binary on install is to release all binaries as a zip file, but that seems like a potentially wasteful download.

**Comment by rberlind** [comment]
I think having OS and ARCH in the names are fine, but we seem to also need version.

**Comment by DRuggeri** [comment]
You're right. I forgot to mention that. I updated the scripts that will produce releases to include the version number in 68e7c24d14f9ed540967bc30977dafa35b056e48. The new format is `terraform-provider-restapi-$tag-$GOOS-$GOARCH`, exactly as you describe.

**Comment by rberlind** [comment]
I forgot to mention that binaries need to be executable.  I don't know if you can set that when generating them or if you need to tell readers to run "chmod +x" against them. Latter does work of course.

**Comment by DRuggeri** [comment]
Right - when using wget/curl/etc, whether the `x` bit is set is determined by your shell's `umask` setting and isn't really something we can control on the server side.

**Comment by rberlind** [comment]
OK.  So, I would just add something to README.md mentioning that people need to use "chmod +x" on the binary.

**Comment by rberlind** [comment]
awesome!
